x/tokenomics/keeper/update_relay_mining_difficulty.go

```diff
@@ -102,6 +101,17 @@ func (k Keeper) UpdateRelayMiningDifficulty(
 			logMessage = fmt.Sprintf("No change in RelayMiningDifficulty for service %s at height %d. Current difficulty: %x", serviceId, sdkCtx.BlockHeight(), newDifficulty.TargetHash)
 		}
 		logger.Info(logMessage)
+
+		// Store the updated difficulty in the map for telemetry.
+		// This is done to only emit the telemetry event if all the difficulties
+		// are updated successfully.
+		difficultyPerServiceMap[serviceId] = newDifficulty
+	}
+
+	for serviceId, newDifficulty := range difficultyPerServiceMap {
```

Comment:
Thoughts on doing this from `abci.go`?

Not sure what the "more pragmatic" location for this code is so open to either approach.